umbraco/Umbraco-CMS
# Netcore: migrate relation tests + Introduced new bulksave and fixed issue in save

### Details
- Updated nuget packages
- Found an issue in `RelationRepository` when tests was moved to LocalDB instead of SqlCE. 
  - `RelationRepository.Save(IEnumerable<IRelation> relations)` did not populate ids back on relations when using LocalDB/SqlServer
- Migrated `RelationRepositoryTests` and `RelationServiceTests`
  - Minimized the amount of data created in some of the tests. These tests took ~25 sec each on the build server before
  - `RelationTypeRepository.PerformGetAll` did not support running tests with no cache
- Found another issue in other PR https://github.com/umbraco/Umbraco-CMS/pull/9136 about save did create duplicate keys..
  - Fixed by introducing `IRelationService.GetByParentChildAndType`
  - Tests migrated in https://github.com/umbraco/Umbraco-CMS/pull/9136 proffs this solved the issue

### Tests
1. Create doc type with a media data type
1. Create a content of that doc type and set media
1. Check the relation is created under Settings > Relations > Relate Media > Relations
